st2common/st2common/models/db/execution.py

```diff
@@ -142,10 +154,22 @@ def mask_secrets(self, value):
                     },
                 )
 
+        output_schema_result = ActionExecutionDB.result.parse_field_value(
+            result["result"]
+        )
+
+        # accessing parameters marked secret as true in the output_schema
+        # and masking them for the result in action execution API
+        if output_schema_result:
+            for key, spec in six.iteritems(result["action"]["output_schema"]):
+                if spec.get("secret", False):
+                    # TODO: Change to output_schema_result[output_key][key] = MASKED_ATTRIBUTE_VALUE
+                    output_schema_result["result"][key] = MASKED_ATTRIBUTE_VALUE
```

Comment:
The result["result"] here is specific to python action. For shell or workflow action, this is going to return key error. There's an output key defined in the action runner model that says which attribute the action is output to in the result (i.e. result for python, output for workflow, etc.). The output key is defined at https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/runner.py#L70. You can see how it is used to validate output schema at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/action_db.py#L247. The mask secret for output will also need to do this. This also means you will need unit tests for result of different action types.